### PR TITLE
Ecomm 50 crud implementation

### PR DESCRIPTION
**Pull request by BrianVega**
Crud controller implementation with springmvc

### CODE REVIEW

File: src/main/java/org/ecommerce/services/jpa/impl/OrderJpaServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Override
+    public GetOrderResponse findById(Long id) {
+        return new GetOrderResponse(
+                orderRepository.findById(id).map(orderDTOMapper)
+                        .orElseThrow(() -> new EntityNotFound(Error.ENTITY_NOT_FOUND.getDescription()))
+        );
+    }
+
+    private Order buildOrderFromDTO(OrderRequestDTO order) {
```
Comment (algutierrez-griddynamics) [inline src/main/java/org/ecommerce/services/jpa/impl/OrderJpaServiceImpl.java:94]: Please create a separate class to map the dto to the order. Add null checks on the attributes or other kind of validations to make sure a valid order is created. Add unit tests.

File: src/test/java/org/ecommerce/repositories/jpa/OrderRepositoryJpaImplTest.java [hunk 2]
```diff
@@ -16,16 +14,17 @@
...
 import java.util.List;
+import java.util.NoSuchElementException;
 
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.lenient;
 
 @ActiveProfiles("local")
 @SpringBootTest
-class OrderServiceJpaImplTest {
+class OrderRepositoryJpaImplTest {
```
Comment (algutierrez-griddynamics) [inline src/test/java/org/ecommerce/repositories/jpa/OrderRepositoryJpaImplTest.java:24]: Please rename to `OrderJpaRepositoryTest`

Comment (BrianVega) [reply src/test/java/org/ecommerce/repositories/jpa/OrderRepositoryJpaImplTest.java]: Done https://github.com/algutierrez-griddynamics/e-commerce/pull/51/commits/d21d67c0ff631e4bbb78eca9c1e8b0e5d731589b

File: src/main/java/org/ecommerce/services/jpa/impl/OrderJpaServiceImpl.java (hunk 1 above)
Comment (BrianVega) [reply src/main/java/org/ecommerce/services/jpa/impl/OrderJpaServiceImpl.java]: i just separate the mapper from the service https://github.com/algutierrez-griddynamics/e-commerce/pull/51/commits/a84c805de9883f6ac5f49e02560ad675f8b59195 logic. About the validations; the instructions of the current implementation asked me just about happy paths, that's why i didn't validate anything hehe. Seems like validations, exception handling and tests are part of the next task: `Validations and Exception handling` -> [ticket](https://github.com/users/algutierrez-griddynamics/projects/2/views/1?pane=issue&itemId=85913420)

File: src/main/java/org/ecommerce/models/services/responses/GetAllOrdersResponse.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.ecommerce.dtos.responses.OrderDTO;
+
+import java.util.List;
+
+@Getter
+@AllArgsConstructor
+public class GetAllOrdersResponse {
+    private List<OrderDTO> orders;
```
[the remark this answers is not in the document]
Comment (BrianVega) [reply src/main/java/org/ecommerce/models/services/responses/GetAllOrdersResponse.java:12]: Pagination is planned to be implemented in the following task: `Search endpoints and pagination`. I just created two tickets with the technical requirements requirements of the task. [This](https://github.com/users/algutierrez-griddynamics/projects/2/views/1?pane=issue&itemId=85913430) is for the `Pagination & Metadata`.  and [this](https://github.com/users/algutierrez-griddynamics/projects/2/views/1?pane=issue&itemId=86439907) is for the `Filtering & Sorting`

**Review by algutierrez-griddynamics — APPROVED** [review]

**Pull request merged**